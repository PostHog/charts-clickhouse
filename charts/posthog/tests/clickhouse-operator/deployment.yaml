suite: ClickHouse deployment definition
templates:
  - templates/clickhouse-operator/deployment.yaml

tests:
  - it: should be empty if clickhouse.enabled is set to false
    set:
      clickhouse.enabled: false
    asserts:
      - hasDocuments:
          count: 0

  - it: the manifest should match the snapshot when using default values
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: metadata.namespace override via clickhouse.namespace should work
    set:
      clickhouse.namespace: custom-namespace
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: metadata
          value:
            name: clickhouse-operator
            namespace: custom-namespace
            labels:
              app: clickhouse-operator
              clickhouse.altinity.com/chop: 0.19.0

  - it: metadata.namespace override via Release.Namespace should work
    release:
      namespace: custom-namespace-from-release
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: metadata
          value:
            name: clickhouse-operator
            namespace: custom-namespace-from-release
            labels:
              app: clickhouse-operator
              clickhouse.altinity.com/chop: 0.19.0
