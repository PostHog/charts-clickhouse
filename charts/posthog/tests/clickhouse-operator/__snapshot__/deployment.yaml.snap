the manifest should match the snapshot when using default values:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: clickhouse-operator
      name: clickhouse-operator
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: clickhouse-operator
      template:
        metadata:
          annotations:
            prometheus.io/port: "8888"
            prometheus.io/scrape: "true"
          labels:
            app: clickhouse-operator
        spec:
          containers:
          - env:
            - name: OPERATOR_POD_NODE_NAME
              valueFrom:
                fieldRef:
                  fieldPath: spec.nodeName
            - name: OPERATOR_POD_NAME
              valueFrom:
                fieldRef:
                  fieldPath: metadata.name
            - name: OPERATOR_POD_NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
            - name: OPERATOR_POD_IP
              valueFrom:
                fieldRef:
                  fieldPath: status.podIP
            - name: OPERATOR_POD_SERVICE_ACCOUNT
              valueFrom:
                fieldRef:
                  fieldPath: spec.serviceAccountName
            - name: OPERATOR_CONTAINER_CPU_REQUEST
              valueFrom:
                resourceFieldRef:
                  containerName: clickhouse-operator
                  resource: requests.cpu
            - name: OPERATOR_CONTAINER_CPU_LIMIT
              valueFrom:
                resourceFieldRef:
                  containerName: clickhouse-operator
                  resource: limits.cpu
            - name: OPERATOR_CONTAINER_MEM_REQUEST
              valueFrom:
                resourceFieldRef:
                  containerName: clickhouse-operator
                  resource: requests.memory
            - name: OPERATOR_CONTAINER_MEM_LIMIT
              valueFrom:
                resourceFieldRef:
                  containerName: clickhouse-operator
                  resource: limits.memory
            image: altinity/clickhouse-operator:latest
            imagePullPolicy: Always
            name: clickhouse-operator
            volumeMounts:
            - mountPath: /etc/clickhouse-operator
              name: etc-clickhouse-operator-folder
            - mountPath: /etc/clickhouse-operator/conf.d
              name: etc-clickhouse-operator-confd-folder
            - mountPath: /etc/clickhouse-operator/config.d
              name: etc-clickhouse-operator-configd-folder
            - mountPath: /etc/clickhouse-operator/templates.d
              name: etc-clickhouse-operator-templatesd-folder
            - mountPath: /etc/clickhouse-operator/users.d
              name: etc-clickhouse-operator-usersd-folder
          - image: altinity/metrics-exporter:latest
            imagePullPolicy: Always
            name: metrics-exporter
            volumeMounts:
            - mountPath: /etc/clickhouse-operator
              name: etc-clickhouse-operator-folder
            - mountPath: /etc/clickhouse-operator/conf.d
              name: etc-clickhouse-operator-confd-folder
            - mountPath: /etc/clickhouse-operator/config.d
              name: etc-clickhouse-operator-configd-folder
            - mountPath: /etc/clickhouse-operator/templates.d
              name: etc-clickhouse-operator-templatesd-folder
            - mountPath: /etc/clickhouse-operator/users.d
              name: etc-clickhouse-operator-usersd-folder
          serviceAccountName: clickhouse-operator
          volumes:
          - configMap:
              name: etc-clickhouse-operator-files
            name: etc-clickhouse-operator-folder
          - configMap:
              name: etc-clickhouse-operator-confd-files
            name: etc-clickhouse-operator-confd-folder
          - configMap:
              name: etc-clickhouse-operator-configd-files
            name: etc-clickhouse-operator-configd-folder
          - configMap:
              name: etc-clickhouse-operator-templatesd-files
            name: etc-clickhouse-operator-templatesd-folder
          - configMap:
              name: etc-clickhouse-operator-usersd-files
            name: etc-clickhouse-operator-usersd-folder
